policies/selfdeal/basic.go

```diff
@@ -0,0 +1,81 @@
+package selfdeal
+
+import (
+	"context"
+
+	"github.com/filecoin-project/specs-actors/actors/abi"
+
+	"github.com/filecoin-project/go-storage-miner/apis/node"
+)
+
+type Chain interface {
+	GetChainHead(ctx context.Context) (node.TipSetToken, abi.ChainEpoch, error)
+}
+
+type BasicPolicy struct {
+	api Chain
+
+	// An estimate for the number of blocks between the current chain head and
+	// when a sector should have been proven. Used to compute the self-deal
+	// StartEpoch.
+	provingDelay abi.ChainEpoch
+
+	// The number of epochs for which the self-dealing miner will receive power.
+	duration abi.ChainEpoch
+}
+
+// NewBasicPolicy produces a new fixed duration self-deal policy.
+func NewBasicPolicy(api Chain, delay abi.ChainEpoch, duration abi.ChainEpoch) BasicPolicy {
+	return BasicPolicy{api: api, provingDelay: delay, duration: duration}
+}
+
+// Schedule produces the deal terms for this fixed duration self-deal policy.
+func (p *BasicPolicy) Schedule(ctx context.Context, pieces ...node.PieceWithOptionalDealInfo) (node.DealSchedule, error) {
+	_, epoch, err := p.api.GetChainHead(ctx)
+	if err != nil {
+		return node.DealSchedule{}, err
+	}
+
+	var start *abi.ChainEpoch
+	var end *abi.ChainEpoch
+
+	for _, p := range pieces {
+		if p.DealInfo != nil {
+			if p.DealInfo.DealSchedule.StartEpoch < epoch {
```

Comment:
I'm not sure this shouldn't be `epoch+p.provingDelay`. What happens if the sector isn't proven by the time the self-deal is put on chain? Given that the miner want the minimum amount of risk, it seems they may want `start == end` where end is the maximum deal duration (if one exists). This could be handled by another policy, so this is ok for now.